src/components/ArticleVotes.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchVotesByArticleId } from "../api";
+
+function ArticleVotes({ article, currentVotes }) {
+    
+  const [votes, setVotes] = useState(0);
+  const [error, setError] = useState(null);
+  const [optimisticMsg, setOptimisitcMsg] = useState(null);
+
+  const article_id = article.article_id;
+
+  const handleVote = (vote) => {
+    setVotes(votes + vote);
+    patchVotesByArticleId(article_id, vote)
+      .then(() => {
+        setOptimisitcMsg("Vote Submitted!");
+      })
+      .catch((error) => {
+        setError("Unable to update votes. Please try again.");
+      });
```

Comment:
Votes are working well when the internet is reliable. However we need to do a bit more than set the error to a message when the vote is unsuccessful, we should also provide the user with some sort of visual indication that the vote hasn't gone through, and reset the vote state to what is was before.